src/replace_landsurface_with_BARRA2R_IC.py

```diff
@@ -191,9 +192,9 @@ def swap_land_barra(mask_fullpath, ec_cb_file_fullpath, ic_date):
 
     # Read in the surface temperature data from the archive
     BARRA_FIELDN = 'ts'
-    indir = BARRA_DIR + '1hr/' + BARRA_FIELDN + '/v20231001'
-    barra_files = glob(indir + '/' + BARRA_FIELDN + '*' + yyyy + mm + '*nc')
-    barra_fname = indir + '/' + barra_files[0].split('/')[-1]
+    indir = os.path.join(BARRA_DIR, '1hr', BARRA_FIELDN, 'v20231001')
+    barra_files = glob(os.path.join(indir, BARRA_FIELDN + '*' + yyyy + mm + '*nc'))
+    barra_fname = os.path.join(indir, os.path.basename(barra_files[0]))
```

Comment:
```suggestion
    barra_fname = barra_files[0]
```

Since in the `barra_files` declaration the `indir` is included in the `glob` search, `barra_files` are absolute paths. Therefore, the `os.path.join(indir, os.path.basename...` part is redundant. You are basically removing the `indir` to add it again.